jmx_prometheus_javaagent/src/main/java/io/prometheus/jmx/JavaAgent.java

```diff
@@ -57,6 +59,27 @@ public static void premain(String agentArgument, Instrumentation instrumentation
      server.setHandler(context);
      context.addServlet(new ServletHolder(new MetricsServlet()), "/metrics");
 
-     server.start();
+     final int delayInSecs = collector.getStartDelay();
```

Comment:
This feels like a layering violation, the delay would be better in the JmxCollecter I think